### PR TITLE
12/1/2020 Programming Session

### PR DESCRIPTION
**Pull request by BrandonWerbel**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/ControlPanelSubsystem.java [hunk 1]
```diff
@@ -112,9 +112,9 @@ public void setDetectedColor(Color detectedColor) {
         this.detectedColor = detectedColor;
     }
 
-    // public VictorSPX getCPMoter(){
-        // return cpMotor;
-    // }
+    public VictorSPX getCPMoter(){
```
Comment (ArchdukeTim) [inline src/main/java/frc/robot/subsystems/ControlPanelSubsystem.java:115]: mot**o**r

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java [hunk 2]
```diff
@@ -1,36 +1,31 @@
...
 
     public IntakeSubsystem() {
-        // upperIntakeMotor.setInverted(InvertType.OpposeMaster);
-        intakeSwitchButton.whenActive(new InstantCommand(() -> { 
-            ballsCollected++;
-            System.out.println("Ball count: " + ballsCollected);
-        }));
+        upperIntakeMotor.follow(bottomIntakeMotor);
+        upperIntakeMotor.setInverted(InvertType.OpposeMaster);
+        intakeSwitchButton.whenActive(new InstantCommand(() -> ballsCollected++));
```
Comment (ArchdukeTim) [inline src/main/java/frc/robot/subsystems/IntakeSubsystem.java:24]: This is probably more a design flaw, but the balls collected shouldn't rely on the button. Does `whenActive` happen while held or just when it's pressed once?

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -80,16 +88,13 @@ private void configureButtonBindings() {
...
             .whenPressed(new InstantCommand(() -> shooterSubsystem.activatePiston(), shooterSubsystem))
             .whenInactive(new InstantCommand(() -> shooterSubsystem.lowerPiston(), shooterSubsystem));
 
-        btnIntakeOut
-            .whenHeld(new InstantCommand(() -> intakeSubsystem.spin(0.5), intakeSubsystem))
-            .whenInactive(new InstantCommand(() -> intakeSubsystem.spin(0), intakeSubsystem), true);
-        btnIntakeIn
-            .whenHeld(new InstantCommand(() -> intakeSubsystem.spin(-0.5), intakeSubsystem))
-            .whenInactive(new InstantCommand(() -> intakeSubsystem.spin(0), intakeSubsystem), true);
+        driveSubsystem.setDefaultCommand(new RunCommand(() -> driveSubsystem.drive(speed, rotation), driveSubsystem));
```
Comment (ArchdukeTim) [inline src/main/java/frc/robot/RobotContainer.java:95]: Uh, do these variables even exist? Looks like you've duplicated 77-81 here, but with the old versions

File: src/main/java/frc/robot/commands/ControlPanelCommand.java [hunk 4]
```diff
@@ -29,6 +30,6 @@ public void execute() {
                 ColorMatchResult resultColor = controlPanel.getColorMatcher().matchClosestColor(readColor);
                 controlPanel.setDetectedColor(resultColor.color);
         }
-        // controlPanel.getCPMoter().set(VictorSPXControlMode.Velocity, (double)controlPanel.getSpeed());
+        controlPanel.getCPMoter().set(VictorSPXControlMode.Velocity, (double)controlPanel.getSpeed());
```
Comment (ArchdukeTim) [inline src/main/java/frc/robot/commands/ControlPanelCommand.java:33]: you should call a method on the controlPanel that controls the motor. Don't get the motor and set it here. What if you changed the type of motor or changed how it worked? Then you'll have to change it here as well as any other place that you use it.

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -62,15 +62,23 @@ private void configureButtonBindings() {
         Joystick rightJoystick = new Joystick(1);
         Joystick altJoystick = new Joystick(2);
```
Comment (ErikBoesen) [inline src/main/java/frc/robot/RobotContainer.java:63]: This is the nitpickiest thing ever, but Tim already got to all the good stuff. So with that being said, `joystickRight` and `joystickAlt` might be more consistent with the `btn______` naming scheme you all seem to have settled on.

**Review by ErikBoesen — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java (hunk 2 above)
Comment (ErikBoesen) [reply src/main/java/frc/robot/subsystems/IntakeSubsystem.java:24]: In theory, if this button is a debounced limit switch type thing, this seems ok to me

**Comment by AndrewLester** [comment]
I'm going to close this in favor of #21 . All changes made here have been moved to that PR, and @ArchdukeTim , your comments are addressed. Also, this branch seems to have a few duplicated commits. @BrandonWerbel since this is your PR that I'm closing, please do make suggestions on the new PR, as I'd like to hear your opinion.

Since #21 has a large number of changed files, make sure to click "viewed" on the ones that aren't really pertinent to hide them (anything in the CTRESim-java folder).

**Pull request closed without merging**